src/Gui/Dialogs/DlgAddPropertyVarSet.cpp

```diff
@@ -186,28 +333,44 @@ void DlgAddPropertyVarSet::initializeWidgets(ViewProviderVarSet* viewProvider)
     // printFocusChain(ui->lineEditName);
 }
 
-void DlgAddPropertyVarSet::setTitle()
+bool DlgAddPropertyVarSet::propertyExists(const std::string& name)
 {
-    setWindowTitle(QObject::tr("Add a property to %1").arg(QString::fromStdString(varSet->getFullName())));
+    App::Property* prop = varSet->getPropertyByName(name.c_str());
+    return prop && prop->getContainer() == varSet;
 }
 
-void DlgAddPropertyVarSet::setOkEnabled(bool enabled)
+bool DlgAddPropertyVarSet::isNameValid()
 {
-    QPushButton *okButton = ui->buttonBox->button(QDialogButtonBox::Ok);
-    okButton->setEnabled(enabled);
+    std::string name = ui->lineEditName->text().toStdString();
+
+    return !(name.empty() ||
+             name != Base::Tools::getIdentifier(name) ||
+             App::ExpressionParser::isTokenAConstant(name) ||
+             App::ExpressionParser::isTokenAUnit(name) ||
+             propertyExists(name));
```

Comment:
That ! at the beginning is very easy to miss, wouldn't it be better to move it inside?